src/router.tsx

```diff
@@ -1,24 +1,26 @@
-import { useEffect, useState } from 'react';
+import { useEffect, useState } from "react";
 
-import { MemoryRouter } from './MemoryRouter';
+import { MemoryRouter } from "./MemoryRouter";
 
 // Default export a singleton:
 const memoryRouter = new MemoryRouter();
 export default memoryRouter;
 
 // Overrides the useRouter hook:
 export const useRouter = () => {
-  const [ router, setRouter ] = useState(memoryRouter);
+  const [router, setRouter] = useState(memoryRouter);
 
   useEffect(() => {
     const handleRouteChange = () => {
       // Clone the (mutable) memoryRouter, to ensure we trigger an update
       setRouter({ ...memoryRouter });
     };
-    memoryRouter.events.on('routeChangeComplete', handleRouteChange);
+    memoryRouter.events.on("routeChangeStart", handleRouteChange);
```

Comment:
Seems weird to update the state on this "start" event, and then again on the "complete" event.  Seems like we should only update the state once, right?